services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/query/FacetQueries.java

```diff
@@ -0,0 +1,99 @@
+package org.sagebionetworks.repo.manager.table.query;
+
+import java.util.List;
+
+import org.sagebionetworks.repo.manager.table.FacetModel;
+import org.sagebionetworks.repo.manager.table.FacetTransformer;
+import org.sagebionetworks.repo.model.table.FacetColumnRequest;
+import org.sagebionetworks.repo.model.table.QueryFilter;
+import org.sagebionetworks.table.cluster.CombinedQuery;
+import org.sagebionetworks.table.cluster.TranslationDependencies;
+import org.sagebionetworks.table.query.ParseException;
+import org.sagebionetworks.table.query.TableQueryParser;
+import org.sagebionetworks.table.query.model.TableExpression;
+
+public class FacetQueries {
+
+	private final List<FacetTransformer> transformers;
+
+	private FacetQueries(List<FacetColumnRequest> selectedFacets, List<QueryFilter> additionalFilters , String originalSql,
+			TranslationDependencies dependencies, Boolean returnFacets) {
+		try {
+			
+			CombinedQuery combined = CombinedQuery.builder()
+					.setQuery(originalSql)
+					.setSchemaProvider(dependencies.getSchemaProvider())
+					// Note: The selected facets must be included in the combined SQL.
```

Comment:
must not maybe? And really, why? -> the FacetModel reapplies the facets again on top of the query so we would end up with double facets